x_clip/dataset.py

```diff
@@ -0,0 +1,66 @@
+import torch
+from pathlib import Path
+from PIL import UnidentifiedImageError, Image
+from torch.utils.data import Dataset
+
+
+class ImageTextDataset(Dataset):
```

Comment:
I think there might be some value to making an independent package with such vision/text datasets (this one and webdataset at least), and depending on it here and in dalle pytorch (and in clip retrieval and probably a bunch of other places)
what do you think?

Reply:
agreed. at this point it's a bit strange that pytorch doesn't have something for this by now - are you aware of anything?

Reply:
besides from discussions at https://github.com/pytorch/pytorch/issues/38419 no I don't know
I think pytorch has been taking the approach of letting the user build their own things on the data side

although as a reminder image+text dataset is something that has begun being useful in 2021, it's not that old yet :)